src/porepy/examples/flow_benchmark_3d_case_2.py

```diff
@@ -0,0 +1,228 @@
+    FractureSolidConstants,
+    Permeability,
+)
+from porepy.fracs.fracture_network_3d import FractureNetwork3d
+
+solid_constants_conductive = FractureSolidConstants(
+    residual_aperture=1e-4,
+    normal_permeability=1e4,
+    fracture_permeability=1e4,
+)
+
+solid_constants_blocking = FractureSolidConstants(
+    residual_aperture=1e-4,
+    normal_permeability=1e-4,
+    fracture_permeability=1e-4,
+)
+
+
+class Geometry(pp.PorePyModel):
+    """Define the Geometry as specified in Section 5.3 of [1]."""
+
+    def set_geometry(self) -> None:
+        """Create mixed-dimensional grid adn fracture network."""
+
+        # Create mixed-dimensional grid and fracture network.
+        self.mdg, self.fracture_network = benchmark_3d_case_2(
+            refinement_level=self.params.get("refinement_level", 0)
+        )
+        self.nd: int = self.mdg.dim_max()
+
+        # Obtain domain and fracture list directly from the fracture network.
+        self._domain = cast(pp.Domain, self.fracture_network.domain)
+        self._fractures = self.fracture_network.fractures
+
+        # Create projection between local and global coordinates fracture grids.
+        pp.set_local_coordinate_projections(self.mdg)
+
+        # Create well network.
+        self.set_well_network()
+        if len(self.well_network.wells) > 0:
+            # Compute intersections.
+            assert isinstance(self.fracture_network, FractureNetwork3d)
+            pp.compute_well_fracture_intersections(
+                self.well_network, self.fracture_network
+            )
+            # Mesh wells and add fractures + intersection grids to mixed-dimensional
+            # grid along with these grids' new interfaces to fractures
+            self.well_network.mesh(self.mdg)
+
+
+class PermeabilitySpecification(Permeability):
+    """Set up permeability values in the mixed-dimensional grid."""
+
+    def _low_perm_zones(self, sd: pp.Grid) -> np.ndarray:
+        """Helper mask corresponding to low permeability zones in the matrix."""
+
+        # Safeguard against wrong dimensionality.
+        if sd.dim < 3:
+            raise ValueError("_low_perm_zones is only meaningful for 3d.")
+
+        cc = sd.cell_centers
+
+        zone_0 = np.logical_and(cc[0, :] > 0.5, cc[1, :] < 0.5)
+        zone_1 = np.logical_and.reduce(
+            tuple(
+                [
+                    cc[0, :] < 0.75,
+                    cc[1, :] > 0.5,
+                    cc[1, :] < 0.75,
+                    cc[2, :] > 0.5,
+                ]
+            )
+        )
+        zone_2 = np.logical_and.reduce(
+            tuple(
+                [
+                    cc[0, :] > 0.625,
+                    cc[0, :] < 0.75,
+                    cc[1, :] > 0.5,
+                    cc[1, :] < 0.625,
+                    cc[2, :] > 0.5,
+                    cc[2, :] < 0.75,
+                ]
+            )
+        )
+
+        return np.logical_or.reduce(tuple([zone_0, zone_1, zone_2]))
```

Comment:
There is no need to first create a list and then convert it to a tuple.

Reply:
Indeed.